package/kedro_viz/data_access/repositories/catalog.py

```diff
@@ -2,11 +2,44 @@
 centralise access to Kedro data catalog."""
 # pylint: disable=missing-class-docstring,missing-function-docstring,protected-access
 from typing import Optional
+from unittest.mock import patch
 
-from kedro.io import AbstractDataSet, DataCatalog, DataSetNotFoundError
+from kedro.io import (
+    AbstractDataSet,
+    AbstractVersionedDataSet,
+    DataCatalog,
+    DataSetNotFoundError,
+    Version,
+)
 
 from kedro_viz.constants import KEDRO_VERSION
 
+# Kedro-Viz doesn't need to know the close matches for missing datasets,
+# and `difflib.get_close_matches` is too costly if performed repeatedly.
+# Before Kedro 0.18.1, use the `_get_dataset` method without the verbose
+# error introduced in https://github.com/kedro-org/kedro/commit/f7dd247.
+if KEDRO_VERSION.match(">=0.16.0") and KEDRO_VERSION.match("<0.18.1"):
+
+    def _get_dataset(
+        self, data_set_name: str, version: Version = None
+    ) -> AbstractDataSet:
+        if data_set_name not in self._data_sets:
+            raise DataSetNotFoundError(
+                "DataSet '{}' not found in the catalog".format(data_set_name)
+            )
+
+        data_set = self._data_sets[data_set_name]
+        if version and isinstance(
+            data_set, AbstractVersionedDataSet
+        ):  # pragma: no cover
```

Comment:
This condition is unreachable in Kedro-Viz, but I left it to be aligned with the original, linked code.